lib/main.coffee

```diff
@@ -351,10 +351,10 @@ module.exports = Hydrogen =
             @wsKernelPicker = new WSKernelPicker (kernel) =>
                 @clearResultBubbles()
 
-                grammar = @editor.getGrammar()
+                grammar = kernel.grammar
```

Comment:
I haven't reviewed the whole PR yet, but here are my thoughts on this point.

I think this discussion has uncovered a design flaw in `KernelManager`:
- running kernels are indexed by grammar language
- when a kernel is created, the kernel is assigned the current `@editor.getGrammar()`
- when a kernel is indexed as a running kernel, the kernel is indexed by the language corresponding to the current `@editor.getGrammar()`

Since the time of creation and the time of registration as running kernel are different, it is possible that those values of `@editor.getGrammar()` may be inconsistent.

The call to `@wsKernelPicker.toggle @editor.getGrammar()` makes me think that they must be consistent, but I'm not certain. @lgeiger , can you trigger this inconsistency?

If there is no way to trigger this inconsistency `kernel.grammar` seems more appropriate to me. But I think we should look at removing `kernel.grammar` to avoid any inconsistencies (this may be difficult because of the way we handle watches).